addons/account/models/company.py

```diff
@@ -651,48 +654,58 @@ def _check_hash_integrity(self):
                 'sequence_number',
                 'secure_sequence_number',
             ]
-            moves = self.env['account.move'].sudo().search_fetch(
-                domain=[
-                    ('journal_id', '=', journal.id),
-                    ('inalterable_hash', '!=', False),
-                ],
-                field_names=field_names,
-                order="secure_sequence_number ASC NULLS LAST, sequence_prefix, sequence_number ASC"
-            )
-
-            if not moves:
-                results.append({
-                    'journal_name': journal.name,
-                    'restricted_by_hash_table': 'V',
-                    'status': 'no_data',
-                    'msg_cover': _('There is no journal entry flagged for accounting data inalterability yet.'),
-                })
-                continue
 
+            offset = 0
+            stop = False
             prefix2result = defaultdict(lambda: {
                 'first_move': self.env['account.move'],
                 'last_move': self.env['account.move'],
                 'corrupted_move': self.env['account.move'],
             })
             last_move = self.env['account.move']
-            current_hash_version = 1
-            for move in moves:
-                prefix_result = prefix2result[move.sequence_prefix]
-                if prefix_result['corrupted_move']:
+            while not stop:
+                moves = self.env['account.move'].sudo().search_fetch(
+                    domain=[
+                        ('journal_id', '=', journal.id),
+                        ('inalterable_hash', '!=', False),
+                    ],
+                    field_names=field_names,
+                    order="secure_sequence_number ASC NULLS LAST, sequence_prefix, sequence_number ASC",
+                    limit=INTEGRITY_HASH_BATCH_SIZE,
+                    offset=offset,
+                )
+                if len(moves) < INTEGRITY_HASH_BATCH_SIZE:
+                    stop = True
+                else:
+                    offset += INTEGRITY_HASH_BATCH_SIZE
+
+                if not moves:
```

Comment:
What if there were moves in previous iterations? (if the total number is a multiple of `INTEGRITY_HASH_BATCH_SIZE`)

Reply:
Indeed 🤦🏻‍♂️, I've added more cases in the test